pkgs/upgrade-maps/default.nix

```diff
@@ -0,0 +1,15 @@
+{
+  auto = {
+    "nodejs-14.21-m1.1" = "nodejs-14.21-m1.2";
+    "nodejs-16.18-m1.1" = "nodejs-16.18-m1.2";
+    "nodejs-18.12-m1.1" = "nodejs-18.12-m1.2";
+    "nodejs-19.1-m1.1" = "nodejs-19.1-m1.2";
+  };
+
+  recommend = {
+    "nodejs-14.21-m1.2" = "nodejs-14.21-m2.0";
+    "nodejs-16.18-m1.2" = "nodejs-16.18-m2.0";
+    "nodejs-18.12-m1.2" = "nodejs-18.12-m2.0";
+    "nodejs-19.1-m1.2" = "nodejs-19.1-m2.0";
+  };
+}
```

Comment:
If we wrote this like:

```nix
{
    "nodejs-14.21-m1.1" = {to = "nodejs-14.21-m1.2"; auto = true;};
    "nodejs-16.18-m1.1" = {to = "nodejs-16.18-m1.2";  auto = true;};
    "nodejs-18.12-m1.1" = {to = "nodejs-18.12-m1.2"; auto = true;};
    "nodejs-19.1-m1.1" = {to = "nodejs-19.1-m1.2"; auto = true;};
    "nodejs-14.21-m1.2" = {to = "nodejs-14.21-m2.0"; };
    "nodejs-16.18-m1.2" = {to = "nodejs-16.18-m2.0"; };
    "nodejs-18.12-m1.2" = {to = "nodejs-18.12-m2.0"; };
    "nodejs-19.1-m1.2" = {to = "nodejs-19.1-m2.0"; };
  };
}
```

Then nix would enforce the requirement that each module can only have one place it goes to.